ironfish/src/chainProcessor.ts

```diff
@@ -73,32 +73,30 @@ export class ChainProcessor {
       `Chain processor head not found in chain: ${this.hash.toString('hex')}`,
     )
 
-    const { fork, isLinear } = await this.chain.findFork(head, chainHead)
-    if (!fork) {
-      return { hashChanged: false }
-    }
+    const { fork } = await this.chain.findFork(head, chainHead)
```

Comment:
What is the fork here in the case that chainHead is ahead? This is ok?

Reply:
Fork will return the smaller of the two if they are on the same linear branch